tests/Integration/Behaviour/Features/Context/Domain/ContactFeatureContext.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit_Framework_Assert;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Command\AddContactCommand;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Command\EditContactCommand;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\ContactNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Query\GetContactForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Contact\QueryResult\EditableContact;
+use PrestaShopBundle\Utils\BoolParser;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use function PrestaShopBundle\Utils\BoolParser;
+
+class ContactFeatureContext extends AbstractDomainFeatureContext
+{
+    private const DEFAULT_LOCALE_ID = 1; // EN locale
+    private const DUMMY_CONTACT_ID = 1;
+
+    /**
+     * @var int
+     */
+    private $defaultLangId;
+
+    public function __construct()
+    {
+        $configuration = CommonFeatureContext::getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
+    }
+
+    /**
+     * @Given there is no contact with id :contactId
+     *
+     * @param int $contactId
+     */
+    public function thereIsNoContactWithId(int $contactId)
+    {
+        try {
+            $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+        } catch (ContactNotFoundException $exception) {
+        }
+    }
+
+    /**
+     * @Given there is contact with id :contactId
+     *
+     * @param int $contactId
+     */
+    public function thereIsContactIsWithId(int $contactId)
+    {
+        $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+    }
+
+    /**
+     * @Then I should be able to get contact with id :contactId for editing
+     *
+     * @param int $contactId
+     */
+    public function iShouldBeAbleToGetContactWithIdForEditing(int $contactId)
+    {
+        $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+    }
+
+    /**
+     * @When I add new contact with the following properties:
+     *
+     * @param TableNode $table
+     */
+    public function iAddNewContactWithTheFollowingProperties(TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        /** @var EditableContact $editablContact */
+        $editableContact = $this->mapToEditableContact(self::DUMMY_CONTACT_ID, $data);
+
+        $addContactCommand = new AddContactCommand(
+            $editableContact->getLocalisedTitles(), $editableContact->isMessagesSavingEnabled()
```

Comment:
```suggestion
            $editableContact->getLocalisedTitles(),
            $editableContact->isMessagesSavingEnabled()
```
each argument should be in separate line